src/data/list/basic.lean

```diff
@@ -3139,6 +3139,53 @@ instance decidable_suffix [decidable_eq α] : ∀ (l₁ l₂ : list α), decidab
     decidable_of_iff' (l₁ = drop (len2-len1) l₂) suffix_iff_eq_drop
   else is_false $ λ h, hl $ length_le_of_sublist $ sublist_of_suffix h
 
+lemma prefix_take_le_iff (L : list (list (option α))) (m n : ℕ) (hm : m < L.length) :
```

Comment:
```suggestion
lemma prefix_take_le_iff {L : list (list (option α))} {m n : ℕ} (hm : m < L.length) :
```
And similar below. For iff statements, we generally use implicit arguments for everything that appears in the iff statement. When this is equivalent to "all arguments implicit," it's convenient to write `prefix_take_le_iff.mp`. That doesn't work here (since you need to give `hm`) but still good practice.

Regardless of special treatment for iff, `L` and `m` can be inferred from `hm`.

Reply:
I incorporated your suggestion, but what do you think about the `n` argument? It can't be inferred easily -- that is why I had to change the proof to use `specialize @IH`.

Reply:
The way you're using IH in the proof is a little "atypical," I'd expect that this lemma will be used more often as a rewrite rule or with `mp`/`mpr`, in which case `n` should usually be inferable. I'd suggest keeping it implicit and changing it back if you find yourself having to `@` this lemma often.